content/docs/document-lifecycle.md

```diff
@@ -0,0 +1,60 @@
+---
+title: 'Document lifecycle'
+weight: 300
+---
+
+A `LangiumDocument` goes through seven different states during its lifecycle:
+1. `Parsed` when an AST has been generated from the content of the document.
+2. `IndexedContent` when the AST nodes have been processed by the `IndexManager`.
+3. `Processed` when the pre-processing steps have been completed.
+4. `Linked` when the `Linker` has resolved cross-references.
+5. `IndexedReferences` when the references have been indexed by the `IndexManager`.
+6. `Validated` when the document has been validated by the `DocumentValidator`.
+7. `Changed` when the document has been modified.
+
+The following diagram depicts the services involved in the lifecycle as well as their main responsibilities. More details about the lifecycle can be found below.
+
+![Lifecycle diagram](/assets/doc-lifecycle/document-lifecycle.png)
```

Comment:
Note that in my overview (https://github.com/langium/langium-website/pull/32#pullrequestreview-897223343) I wrote different descriptions of some services that are used multiple times (AstNodeDescriptionProvider and AstNodeLocator). This is because different service methods are used. I did not include the service method names, but we could consider adding them to the diagram.